config/crd/bases/tinkerbell.org_tinkerbell.yaml

```diff
@@ -0,0 +1,241 @@
+                            description: Repository is used to set the image repository
+                              for tinkerbell services.
+                            type: string
+                          tag:
+                            description: Tag is used to set the image tag for tinkerbell
+                              services.
+                            type: string
+                        type: object
+                      trustedProxies:
+                        description: TrustedProxies comma separated allowed CIDRs
+                          subnets to be used as trusted proxies
+                        type: string
+                    required:
+                    - image
+                    - trustedProxies
+                    type: object
+                  tinkController:
+                    description: TinkController contains all the information and spec
+                      about tink controller.
+                    properties:
+                      repository:
+                        description: Repository is used to set the image repository
+                          for tinkerbell services.
+                        type: string
+                      tag:
+                        description: Tag is used to set the image tag for tinkerbell
+                          services.
+                        type: string
+                    type: object
+                  tinkServer:
+                    description: TinkServer contains all the information and spec
+                      about tink server.
+                    properties:
+                      image:
+                        description: Image specifies the details of a tinkerbell services
+                          images
+                        properties:
+                          repository:
+                            description: Repository is used to set the image repository
+                              for tinkerbell services.
+                            type: string
+                          tag:
+                            description: Tag is used to set the image tag for tinkerbell
+                              services.
+                            type: string
+                        type: object
+                      tinkerbellTLS:
+                        description: TinkerbellTLS sets if the tink server should
+                          run with TLS or not.
+                        type: boolean
+                    required:
+                    - image
+                    - tinkerbellTLS
+                    type: object
+                required:
+                - boots
+                - hegel
+                - rufio
+                - tinkController
+                - tinkServer
+                type: object
+              version:
+                description: Version is the Tinkerbell CRD version.
+                type: string
+            required:
+            - clusterDNS
```

Comment:
whats the thought process here around making clusterDNS required?

Reply:
Answered in the next comment.

Reply:
sorry, here i mean can this be optional?

Reply:
Sure thing, will some documentation about it in the how-to section which I am planning to add once I finish the CRD and the propagation of the CRD fields. Usually this is derived from node-local-dns configs or it is customized based on the used dns server whether it is part of the cluster or an external one.